lib/bsdlib/nrf91_sockets.c

```diff
@@ -703,6 +704,42 @@ static int nrf91_socket_offload_getaddrinfo(const char *node,
 	return retval;
 }
 
+static int nrf91_socket_offload_fcntl(int fd, int cmd, va_list args)
+{
+	int retval;
+	int flags;
+
+	switch (cmd) {
+	case F_SETFL:
+		flags = va_arg(args, int);
+		if (flags != 0 && flags != O_NONBLOCK)
+			goto error;
+
+		/* Translate flags from native to nRF */
+		flags = (flags & O_NONBLOCK) ? NRF_O_NONBLOCK : 0;
+
+		retval = nrf_fcntl(fd, NRF_F_SETFL, flags);
+		break;
+
+	case F_GETFL:
+		flags = nrf_fcntl(fd, NRF_F_GETFL, 0);
+
+		/* Translate flags from nRF to native */
+		retval = (flags & NRF_O_NONBLOCK) ? O_NONBLOCK : 0;
+		break;
+
+	default:
+		goto error;
+	}
+
+	return retval;
+
+error:
+	retval = -1;
+	errno = -ENOTSUP;
```

Comment:
`errno` value should not be negated. But I see it's an issue all across the file, we should fix this in a bulk in a separate PR.

And I'd consider `EINVAL` in this case, all in all that's arguemnts that are not correct. `ENOTSUP` is not specified by man pages as an error code for `fcntl`.

Reply:
I followed the existing pattern for error handling in this file without thinking about the negative value returned.  I'll update this PR to return positive EINVAL in this case.